packages/generator/src/generators/template-builders/builder.ts

```diff
@@ -0,0 +1,106 @@
+import {NextConfigComplete} from "next/dist/server/config-shared"
+import {GeneratorOptions} from "../../generator"
+import {
+  addSpaceBeforeCapitals,
+  camelCaseToKebabCase,
+  singleCamel,
+  singlePascal,
+} from "../../utils/inflector"
+
+export interface IBuilder<T> {
+  getTemplateValues(Options: T): Promise<any>
```

Comment:
I know it's not the subject of this PR, but wondering if we can narrow it. From what I see `getTemplateValues` mostly returns a promise with something like:
```
    return {
      name: this.options.name,
      Name: this.options.Name,
    }
```